### PR TITLE
Update aws-batch CE page with EFS and FSx mount path details

### PR DESCRIPTION
**Pull request by llewellyn-sl**
Added explainer text to EFS and FSx steps in AWS Batch CE page.

### CODE REVIEW

**Comment by gwright99** [comment]
@abhi18av - I  don't think this change reflects what we were thinking when we opened: https://github.com/seqeralabs/tower-backlog/issues/719

If I recall correctly, we thought that the mount points should always be specifically defined - even if the same as the already-defined workdir field. (_see Slack snippet_). Agree?

**Comment by llewellyn-sl** [comment]
> @abhi18av - I don't think this change reflects what we were thinking when we opened: [seqeralabs/tower-backlog#719](https://github.com/seqeralabs/tower-backlog/issues/719)
> 
> If I recall correctly, we thought that the mount points should always be specifically defined - even if the same as the already-defined workdir field. (_see Slack snippet_). Agree?

@gwright99 I agree. I initially linked the PR to [seqeralabs/tower-backlog#719](https://github.com/seqeralabs/tower-backlog/issues/719), but removed the link last night since this PR does not address specifying sensible default mount points. This PR just addresses the hole in the docs in terms of describing the actual UI fields and behavior when creating/using EFS and FSx  file systems.

File: docs/compute-envs/aws-batch.md [hunk 1]
```diff
@@ -143,11 +143,15 @@ Once the AWS resources are set up, we can add a new **AWS Batch** environment in
 
 15. Enter any additional **Allowed S3 buckets** that your workflows require to read input data or write output data. The **Pipeline work directory** bucket above is added by default to the list of **Allowed S3 buckets**.
 
-16. To use **EFS**, you can either select **Use existing EFS file system** and specify an existing EFS instance or select **Create new EFS file system** to create one automatically.
+16. To use **EFS**, you can either select **Use existing EFS file system** and specify an existing EFS instance, or select **Create new EFS file system** to create one.
 
-17. To use **FSx**, set the **FSx mount path** to `/fsx` and set the **Pipeline work directory** to `/fsx/work`.
+    - To use an existing EFS file system, enter the **EFS file system id** and **EFS mount path**. If left blank, the mount path defaults to the pipeline work directory root path. This must be a valid mount path for an existing EFS file system that is accessible to your environment.
```
Comment (jordeu) [inline docs/compute-envs/aws-batch.md:148]: For me, this sentence _"This must be a valid mount path for an existing EFS file system that is accessible to your environment."_ is a bit confusing. I think that here we just need to stress that this path is where the EFS volume will be available inside the compute containers.

File: docs/compute-envs/aws-batch.md [hunk 2]
```diff
@@ -143,11 +143,15 @@ Once the AWS resources are set up, we can add a new **AWS Batch** environment in
...
+    - To create a new EFS file system, enter the **EFS mount path**. If left blank, the pipeline work directory will be used as the new EFS file system mount path. 
 
-    ![](_images/aws_lustre_options.png)
+17. To use **FSx for Lustre**, you can either select **Use existing FSx file system** and specify an existing FSx instance, or select **Create new FSx file system** to create one.
+
+- To use an existing FSx file system, enter the **FSx DNS name** and **FSx mount path**. If left blank, the mount path defaults to the pipeline work directory root path. This must be a valid mount path for an existing FSx file system that is accessible to your environment.
```
Comment (jordeu) [inline docs/compute-envs/aws-batch.md:153]: Same comment here at _"This must be a valid mount path for an existing FSx file system that is accessible to your environment."_

**Comment by gwright99** [comment]
> I consider vague guidance / implicit inference to be user experience footguns.

I think we would be well served with more prescriptive guidance here. Customers don't need to heed it, but it makes our position much clearer. Eg.

"Seqera advises the following default mount paths for clarity:
- /mnt/efs/
- /mnt/fsx

If you wish to use EFS or FSX as your work directory, please ensure you work directory matches this value"

Alternatively we new FE widget that allows the user to specify that EFS / FSX are meant to be the work dir rather than S3 and this updates the workdir field accordingly.

File: docs/compute-envs/aws-batch.md (hunk 1 above)
Comment (llewellyn-sl) [inline docs/compute-envs/aws-batch.md:148]:
```suggestion
    - To use an existing EFS file system, enter the **EFS file system id** and **EFS mount path**. This is the path where the EFS volume is accessible to the compute environment. If left blank, the mount path defaults to the pipeline work directory root path.
```

Comment (gwright99) [reply docs/compute-envs/aws-batch.md:148]: IMO the generic text provides less useful guidance. I'm also not 100% that leaving it blank defaults to the work dir root path (see the support ticket Abhinav and I referenced - I seem to recall the default mount being something else).

File: docs/compute-envs/aws-batch.md (hunk 2 above)
Comment (llewellyn-sl) [inline docs/compute-envs/aws-batch.md:153]:
```suggestion
- To use an existing FSx file system, enter the **FSx DNS name** and **FSx mount path**. The FSx mount path is the path where the FSx volume is accessible to the compute environment. If left blank, the mount path defaults to the pipeline work directory root path. 
```

File: docs/compute-envs/aws-batch.md [hunk 3]
```diff
@@ -143,11 +143,15 @@ Once the AWS resources are set up, we can add a new **AWS Batch** environment in
 
 15. Enter any additional **Allowed S3 buckets** that your workflows require to read input data or write output data. The **Pipeline work directory** bucket above is added by default to the list of **Allowed S3 buckets**.
 
-16. To use **EFS**, you can either select **Use existing EFS file system** and specify an existing EFS instance or select **Create new EFS file system** to create one automatically.
+16. To use **EFS**, you can either select **Use existing EFS file system** and specify an existing EFS instance, or select **Create new EFS file system** to create one.
 
-17. To use **FSx**, set the **FSx mount path** to `/fsx` and set the **Pipeline work directory** to `/fsx/work`.
+    - To use an existing EFS file system, enter the **EFS file system id** and **EFS mount path**. This is the path where the EFS volume is accessible to the compute environment. If left blank, the mount path defaults to the pipeline work directory root path.
```
Comment (llewellyn-sl) [inline docs/compute-envs/aws-batch.md:148]:
```suggestion
    - To use an existing EFS file system, enter the **EFS file system id** and **EFS mount path**. This is the path where the EFS volume is accessible to the compute environment. For simplicity, we advise that you specify `/mnt/efs` as the EFS mount path.
```

File: docs/compute-envs/aws-batch.md [hunk 4]
```diff
@@ -143,11 +143,15 @@ Once the AWS resources are set up, we can add a new **AWS Batch** environment in
...
+    - To create a new EFS file system, enter the **EFS mount path**. If left blank, the pipeline work directory will be used as the new EFS file system mount path.
```
Comment (llewellyn-sl) [inline docs/compute-envs/aws-batch.md:149]:
```suggestion
    - To create a new EFS file system, enter the **EFS mount path**. We advise that you specify `/mnt/efs` as the EFS mount path.
```

File: docs/compute-envs/aws-batch.md [hunk 5]
```diff
@@ -143,11 +143,15 @@ Once the AWS resources are set up, we can add a new **AWS Batch** environment in
...
 
-17. To use **FSx**, set the **FSx mount path** to `/fsx` and set the **Pipeline work directory** to `/fsx/work`.
+    - To use an existing EFS file system, enter the **EFS file system id** and **EFS mount path**. This is the path where the EFS volume is accessible to the compute environment. If left blank, the mount path defaults to the pipeline work directory root path.
+    - To create a new EFS file system, enter the **EFS mount path**. If left blank, the pipeline work directory will be used as the new EFS file system mount path. 
 
-    ![](_images/aws_lustre_options.png)
+17. To use **FSx for Lustre**, you can either select **Use existing FSx file system** and specify an existing FSx instance, or select **Create new FSx file system** to create one.
+
+- To use an existing FSx file system, enter the **FSx DNS name** and **FSx mount path**. The FSx mount path is the path where the FSx volume is accessible to the compute environment. If left blank, the mount path defaults to the pipeline work directory root path.
```
Comment (llewellyn-sl) [inline docs/compute-envs/aws-batch.md:153]:
```suggestion
- To use an existing FSx file system, enter the **FSx DNS name** and **FSx mount path**. The FSx mount path is the path where the FSx volume is accessible to the compute environment. For simplicity, we advise that you specify `/mnt/fsx` as the FSx mount path.
```

File: docs/compute-envs/aws-batch.md [hunk 6]
```diff
@@ -143,11 +143,15 @@ Once the AWS resources are set up, we can add a new **AWS Batch** environment in
...
+    - To use an existing EFS file system, enter the **EFS file system id** and **EFS mount path**. This is the path where the EFS volume is accessible to the compute environment. If left blank, the mount path defaults to the pipeline work directory root path.
+    - To create a new EFS file system, enter the **EFS mount path**. If left blank, the pipeline work directory will be used as the new EFS file system mount path. 
 
-    ![](_images/aws_lustre_options.png)
+17. To use **FSx for Lustre**, you can either select **Use existing FSx file system** and specify an existing FSx instance, or select **Create new FSx file system** to create one.
+
+- To use an existing FSx file system, enter the **FSx DNS name** and **FSx mount path**. The FSx mount path is the path where the FSx volume is accessible to the compute environment. If left blank, the mount path defaults to the pipeline work directory root path. 
+- To create a new FSx file system, enter the **FSx size** (in GB) and the **FSx mount path**. If left blank, the pipeline work directory will be used as the new FSx file system mount path.
```
Comment (llewellyn-sl) [inline docs/compute-envs/aws-batch.md:154]:
```suggestion
- To create a new FSx file system, enter the **FSx size** (in GB) and the **FSx mount path**. We advise that you specify `/mnt/fsx` as the FSx mount path.
```

File: docs/compute-envs/aws-batch.md [hunk 7]
```diff
@@ -143,11 +143,15 @@ Once the AWS resources are set up, we can add a new **AWS Batch** environment in
 
 15. Enter any additional **Allowed S3 buckets** that your workflows require to read input data or write output data. The **Pipeline work directory** bucket above is added by default to the list of **Allowed S3 buckets**.
 
-16. To use **EFS**, you can either select **Use existing EFS file system** and specify an existing EFS instance or select **Create new EFS file system** to create one automatically.
+16. To use **EFS**, you can either select **Use existing EFS file system** and specify an existing EFS instance, or select **Create new EFS file system** to create one.
 
-17. To use **FSx**, set the **FSx mount path** to `/fsx` and set the **Pipeline work directory** to `/fsx/work`.
+    - To use an existing EFS file system, enter the **EFS file system id** and **EFS mount path**. This is the path where the EFS volume is accessible to the compute environment. For simplicity, we advise that you specify `/mnt/efs` as the EFS mount path.
```
Comment (llewellyn-sl) [inline docs/compute-envs/aws-batch.md:148]:
```suggestion
    - To use an existing EFS file system, enter the **EFS file system id** and **EFS mount path**. This is the path where the EFS volume is accessible to the compute environment. For simplicity, we advise that you use `/mnt/efs` as the EFS mount path.
```

File: docs/compute-envs/aws-batch.md [hunk 8]
```diff
@@ -143,11 +143,15 @@ Once the AWS resources are set up, we can add a new **AWS Batch** environment in
...
+    - To create a new EFS file system, enter the **EFS mount path**. We advise that you specify `/mnt/efs` as the EFS mount path.
 
-    ![](_images/aws_lustre_options.png)
+17. To use **FSx for Lustre**, you can either select **Use existing FSx file system** and specify an existing FSx instance, or select **Create new FSx file system** to create one.
+
+- To use an existing FSx file system, enter the **FSx DNS name** and **FSx mount path**. The FSx mount path is the path where the FSx volume is accessible to the compute environment. For simplicity, we advise that you specify `/mnt/fsx` as the FSx mount path.
```
Comment (llewellyn-sl) [inline docs/compute-envs/aws-batch.md:153]:
```suggestion
- To use an existing FSx file system, enter the **FSx DNS name** and **FSx mount path**. The FSx mount path is the path where the FSx volume is accessible to the compute environment. For simplicity, we advise that you use `/mnt/fsx` as the FSx mount path.
```

**Comment by llewellyn-sl** [comment]
@gwright99 @jordeu Please re-review this content. I updated it to include a recommended mount path, per Graham's suggestion. Let me know if this reads as accurate, as there seems to be differing views on what default is used if the mount path is left blank.

**Review by jordeu — APPROVED** [review]

**Comment by gwright99** [comment]
> @gwright99 @jordeu Please re-review this content. I updated it to include a recommended mount path, per Graham's suggestion. Let me know if this reads as accurate, as there seems to be differing views on what default is used if the mount path is left blank.

Explicitly suggested mount paths looks good to me. 

There's a slight complication in that - if the client wants to use EFS / FSX as their workdir -  the  POSIX path needs to be specified in the **Pipeline work directory** field higher up on the form (which should mirror the mount path + (optional) subdirectories). 

We don't specifically call this out in the EFS/FSX instructions. However, adding it could make the instructions more complicated. I'm 50-50 whether it's good enough or still needs more guidance. Thoughts?

**Comment by llewellyn-sl** [comment]
> There's a slight complication in that - if the client wants to use EFS / FSX as their workdir - the POSIX path needs to be >specified in the Pipeline work directory field higher up on the form (which should mirror the mount path + (optional) >subdirectories).

>We don't specifically call this out in the EFS/FSX instructions.

@gwright99 If there's a discrepancy between the actual behavior and what we describe, we should aim for accuracy and add to it. I'll review the full AWS doc tomorrow to ensure the pipeline work directory caveat is made clear.

**Comment by llewellyn-sl** [comment]
Added pipeline work directory note to EFS and FSx steps for clarity.

**Review by abhi18av — APPROVED** [review]
For me this is looking okay ✅

**Comment by gwright99** [comment]
Oh no! I'm the nitpicker on this PR :(

Essentially, there are a few overarching scenarios the docs need to be mindful of:
1. A CE which uses an S3-based work-directory, with EFS and/or FSx instances mounted as secondary sources (e.g. source of reference files).
2. A CE which will use EFS and/or FSx as its work directory.

In the case of 1, there is no need to modify the **Pipeline work directory** to reflect the EFS or FSx mount. This will actually break things as the S3 bucket will no longer be used as the work directory.

In the casse of 2, a modification of the **Pipeline work directory** is required (mirroring the mount path that was specified while creating/defining one of those network storage solutions.

Agree? Does this make sense?

File: docs/compute-envs/aws-batch.md [hunk 9]
```diff
@@ -143,11 +143,15 @@ Once the AWS resources are set up, we can add a new **AWS Batch** environment in
 
 15. Enter any additional **Allowed S3 buckets** that your workflows require to read input data or write output data. The **Pipeline work directory** bucket above is added by default to the list of **Allowed S3 buckets**.
 
-16. To use **EFS**, you can either select **Use existing EFS file system** and specify an existing EFS instance or select **Create new EFS file system** to create one automatically.
+16. To use **EFS**, you can either select **Use existing EFS file system** and specify an existing EFS instance, or select **Create new EFS file system** to create one. For new and existing EFS file systems, you will need to specify `<your_EFS_mount_path>/work` as your **Pipeline work directory** (step 8 of this guide).
```
Comment (llewellyn-sl) [inline docs/compute-envs/aws-batch.md:146]:
```suggestion
16. To use **EFS**, you can either select **Use existing EFS file system** and specify an existing EFS instance, or select **Create new EFS file system** to create one. If you intend to use the EFS file system as your work directory, you will need to specify `<your_EFS_mount_path>/work` as your **Pipeline work directory** (step 8 of this guide).
```

File: docs/compute-envs/aws-batch.md [hunk 10]
```diff
@@ -143,11 +143,15 @@ Once the AWS resources are set up, we can add a new **AWS Batch** environment in
...
 
-17. To use **FSx**, set the **FSx mount path** to `/fsx` and set the **Pipeline work directory** to `/fsx/work`.
+    - To use an existing EFS file system, enter the **EFS file system id** and **EFS mount path**. This is the path where the EFS volume is accessible to the compute environment. For simplicity, we advise that you use `/mnt/efs` as the EFS mount path.
+    - To create a new EFS file system, enter the **EFS mount path**. We advise that you specify `/mnt/efs` as the EFS mount path.
 
-    ![](_images/aws_lustre_options.png)
+17. To use **FSx for Lustre**, you can either select **Use existing FSx file system** and specify an existing FSx instance, or select **Create new FSx file system** to create one. For new and existing FSx file systems, you will need to specify `<your_FSx_mount_path>/work` as your **Pipeline work directory** (step 8 of this guide).
```
Comment (llewellyn-sl) [inline docs/compute-envs/aws-batch.md:151]:
```suggestion
17. To use **FSx for Lustre**, you can either select **Use existing FSx file system** and specify an existing FSx instance, or select **Create new FSx file system** to create one. If you intend to use the FSx file system as your work directory, you will need to specify `<your_FSx_mount_path>/work` as your **Pipeline work directory** (step 8 of this guide). 
```

File: docs/compute-envs/aws-batch.md [hunk 11]
```diff
@@ -143,11 +143,15 @@ Once the AWS resources are set up, we can add a new **AWS Batch** environment in
 
 15. Enter any additional **Allowed S3 buckets** that your workflows require to read input data or write output data. The **Pipeline work directory** bucket above is added by default to the list of **Allowed S3 buckets**.
 
-16. To use **EFS**, you can either select **Use existing EFS file system** and specify an existing EFS instance or select **Create new EFS file system** to create one automatically.
+16. To use **EFS**, you can either select **Use existing EFS file system** and specify an existing EFS instance, or select **Create new EFS file system** to create one. If you intend to use the EFS file system as your work directory, you will need to specify `<your_EFS_mount_path>/work` as your **Pipeline work directory** (step 8 of this guide).
```
Comment (llewellyn-sl) [inline docs/compute-envs/aws-batch.md:146]:
```suggestion
16. To use **EFS**, you can either select **Use existing EFS file system** and specify an existing EFS instance, or select **Create new EFS file system** to create one. If you intend to use the EFS file system as your work directory, you will need to specify `<your_EFS_mount_path>/work` in the **Pipeline work directory** field(step 8 of this guide).
```

File: docs/compute-envs/aws-batch.md [hunk 12]
```diff
@@ -143,11 +143,15 @@ Once the AWS resources are set up, we can add a new **AWS Batch** environment in
...
 
-17. To use **FSx**, set the **FSx mount path** to `/fsx` and set the **Pipeline work directory** to `/fsx/work`.
+    - To use an existing EFS file system, enter the **EFS file system id** and **EFS mount path**. This is the path where the EFS volume is accessible to the compute environment. For simplicity, we advise that you use `/mnt/efs` as the EFS mount path.
+    - To create a new EFS file system, enter the **EFS mount path**. We advise that you specify `/mnt/efs` as the EFS mount path.
 
-    ![](_images/aws_lustre_options.png)
+17. To use **FSx for Lustre**, you can either select **Use existing FSx file system** and specify an existing FSx instance, or select **Create new FSx file system** to create one. If you intend to use the FSx file system as your work directory, you will need to specify `<your_FSx_mount_path>/work` as your **Pipeline work directory** (step 8 of this guide).
```
Comment (llewellyn-sl) [inline docs/compute-envs/aws-batch.md:151]:
```suggestion
17. To use **FSx for Lustre**, you can either select **Use existing FSx file system** and specify an existing FSx instance, or select **Create new FSx file system** to create one. If you intend to use the FSx file system as your work directory, you will need to specify `<your_FSx_mount_path>/work` in the **Pipeline work directory** field (step 8 of this guide). 
```

**Comment by llewellyn-sl** [comment]
> Oh no! I'm the nitpicker on this PR :(
> 
> Essentially, there are a few overarching scenarios the docs need to be mindful of:
> 
> 1. A CE which uses an S3-based work-directory, with EFS and/or FSx instances mounted as secondary sources (e.g. source of reference files).
> 2. A CE which will use EFS and/or FSx as its work directory.
> 
> In the case of 1, there is no need to modify the **Pipeline work directory** to reflect the EFS or FSx mount. This will actually break things as the S3 bucket will no longer be used as the work directory.
> 
> In the case of 2, a modification of the **Pipeline work directory** is required (mirroring the mount path that was specified while creating/defining one of those network storage solutions).
> 
> Agree? Does this make sense?

Important nuance, thanks for flagging. Done.

**Pull request merged**